rust/operator-binary/src/catalog/commons.rs

```diff
@@ -65,6 +65,18 @@ impl ExtendCatalogConfig for MetastoreConnection {
         // This is tightly coupled with the hive discovery config map data layout now
         let transformed_hive_connection = hive_connection.replace('\n', ",");
 
+        // NOTE (@NickLarsenNZ): This enables Legacy S3 support via the
+        // `hive.s3.*` properties.
+        // This will be removed in a future version of Trino.
+        // See: https://trino.io/docs/469/object-storage/legacy-s3.html
+        catalog_config.add_property("fs.hadoop.enabled", "true");
+
+        // NOTE (@NickLarsenNZ): To migrate a catalog from Legacy S3 to Native
+        // S3, `fs.native-s3.enabled=true` needs to be set, and the `hive.s3.*`
+        // properties need to become `s3.*`.
+        // Presumably, `fs.hadoop.enabled` can be removed afterwards.
+        // See: https://trino.io/docs/469/object-storage/legacy-s3.html#migration-to-s3-file-system
```

Comment:
@sbernauer, any opinions on this?

See also: https://github.com/stackabletech/docker-images/issues/971#issuecomment-2634502954

Reply:
I personally feel like we should go with the new native implementation by default (so set `fs.native-s3.enabled` and use `s3.*` instead of `hive.s3.*`) for 469 and higher. And yes, with the current restrictions (which means e,g, to snafu Error when S3 doesn't use TLS and such)

What we can do for the user is add some docs how they can use configOverrides on the catalog to switch back to the legacy implementation (with a disclaimer that Trino can remove it at any point).

That being said: I just came back from vacation, totally happy to discuss this offline!

Reply:
As discussed, here is the refactor that can handle different behaviours for different ranges of versions.

https://github.com/stackabletech/trino-operator/compare/9a695e32394339569b78b1a75ae8c86750d39ecc..5b6b45af17094e5c4f1af51885b1d136f1481336

Reply:
Many thanks, I reviewed the PR